controllers/operatorconfigmap_controller.go

```diff
@@ -0,0 +1,484 @@
+/*
+Copyright 2023 Red Hat OpenShift Data Foundation.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+
+	// The embed package is required for the prometheus rule files
+	_ "embed"
+
+	"github.com/red-hat-storage/ocs-client-operator/pkg/console"
+	"github.com/red-hat-storage/ocs-client-operator/pkg/csi"
+	"github.com/red-hat-storage/ocs-client-operator/pkg/templates"
+
+	"github.com/go-logr/logr"
+	configv1 "github.com/openshift/api/config/v1"
+	secv1 "github.com/openshift/api/security/v1"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	k8sYAML "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+//go:embed pvc-rules.yaml
+var pvcPrometheusRules string
+
+const (
+	operatorConfigMapName = "ocs-client-operator-manager-config"
+	// ClusterVersionName is the name of the ClusterVersion object in the
+	// openshift cluster.
+	clusterVersionName = "version"
+
+	OperatorConfigMapFinalizer = "operatorcm.ocs.openshift.io"
```

Comment:
`operatorcm.ocs.openshift.io` doesn't sound better as `operatorcm` seems generic, maybe exact one `ocs-client-operator.ocs.openshift.io` might be better, usually these names take after their CRs (but i believe storageclient & storageclassclaim is already taken by their corresponding controllers)

I don't think we need to export the constant, don't capitalize the var